Makefile

```diff
@@ -15,150 +15,145 @@
 #
 
 MAJOR := 0
-MINOR := 16
-PATCH := 2
-LABEL := -test2
+MINOR := 17
+PATCH := 0
+LABEL := -test1
 VERSION := $(MAJOR).$(MINOR).$(PATCH)$(LABEL)
 
-TOOLS_DEB    := machine-emulator-tools-v$(VERSION).deb
-TOOLS_IMAGE  := cartesi/machine-emulator-tools:$(VERSION)
+TOOLS_TARGZ  := machine-guest-tools-v$(VERSION).tar.gz
+TOOLS_IMAGE  := cartesi/machine-guest-tools:$(VERSION)
 TOOLS_ROOTFS := rootfs-tools-v$(VERSION).ext2
 TOOLS_ROOTFS_IMAGE := cartesi/rootfs-tools:$(VERSION)
 
-IMAGE_KERNEL_VERSION ?= v0.20.0
+LINUX_IMAGE_VERSION ?= v0.20.0
 LINUX_VERSION ?= 6.5.13-ctsi-1
-LINUX_HEADERS_URLPATH := https://github.com/cartesi/image-kernel/releases/download/${IMAGE_KERNEL_VERSION}/linux-libc-dev-riscv64-cross-${LINUX_VERSION}-${IMAGE_KERNEL_VERSION}.deb
+LINUX_HEADERS_URLPATH := https://github.com/cartesi/machine-linux-image/releases/download/${LINUX_IMAGE_VERSION}/linux-libc-dev-riscv64-cross-${LINUX_VERSION}-${LINUX_IMAGE_VERSION}.deb
+LINUX_HEADERS_SHA256 := 2723435e8b45d8fb7a79e9344f6dc517b3dbc08e03ac17baab311300ec475c08
 
-all: fs
+PREFIX ?= /usr
+DESTDIR ?= .
+CARGO ?= cargo
+ARCH = $(shell uname -m)
 
-build: control package.json
+ifneq ($(ARCH),riscv64)
+all: build
+else
+all: build-riscv64 ## Build all tools
+endif
+
+build-riscv64: sys-utils rollup-http package.json ## Build riscv64 tools
+
+build: targz fs ## Build targz and fs (cross compiling with Docker)
+
+targz: ## Build tools .tar.gz (cross compiling with Docker)
 	@docker buildx build --load \
-		--build-arg TOOLS_DEB=$(TOOLS_DEB) \
-		--build-arg IMAGE_KERNEL_VERSION=$(IMAGE_KERNEL_VERSION) \
-		--build-arg LINUX_VERSION=$(LINUX_VERSION) \
+		--build-arg TOOLS_TARGZ=$(TOOLS_TARGZ) \
 		--build-arg LINUX_HEADERS_URLPATH=$(LINUX_HEADERS_URLPATH) \
+		--build-arg LINUX_HEADERS_SHA256=$(LINUX_HEADERS_SHA256) \
 		-t $(TOOLS_IMAGE) \
 		-f Dockerfile \
 		. ;
 	$(MAKE) copy
 
 copy:
 	@ID=`docker create $(TOOLS_IMAGE)` && \
-	   docker cp $$ID:/opt/cartesi/$(TOOLS_DEB) . && \
+	   docker cp $$ID:/work/$(TOOLS_TARGZ) . && \
 	   docker rm $$ID
 
-$(TOOLS_DEB) deb: build
-
-control: Makefile control.in
-	@sed 's|ARG_VERSION|$(VERSION)|g' control.in > control
 package.json: Makefile package.json.in
 	@sed 's|ARG_VERSION|$(VERSION)|g' package.json.in > package.json
 
-fs: $(TOOLS_ROOTFS)
+fs: $(TOOLS_ROOTFS) ## Build tools rootfs image (cross compiling with Docker)
 
-$(TOOLS_ROOTFS): $(TOOLS_DEB)
+$(TOOLS_ROOTFS): $(TOOLS_TARGZ) fs/Dockerfile
 	@docker buildx build --platform linux/riscv64 \
-	  --build-arg TOOLS_DEB=$(TOOLS_DEB) \
+	  --build-arg TOOLS_TARGZ=$(TOOLS_TARGZ) \
 	  --output type=tar,dest=rootfs.tar \
 	  --file fs/Dockerfile \
 	  . && \
-	xgenext2fs -fzB 4096 -b 25600 -i 4096 -a rootfs.tar -L rootfs $(TOOLS_ROOTFS) && \
+	xgenext2fs -fzB 4096 -i 4096 -r +4096 -a rootfs.tar -L rootfs $(TOOLS_ROOTFS) && \
```

Comment:
I don't really remember why but someone asked for this image to be a fixed size.
@vfusco do you remember this detail?

Reply:
I changed because when updating to Ubuntu 24.04 I got some very strange errors, took me a while to understand that the size was not big enough. 

But this image is going to be removed, or used only for CI testing, so I don't see a problem on the change, packages won't use this image.

Although it won't be used to packages, we have to decide if we keep tools a rootfs image so we can test new tools release in our CI more easily, if so maybe then rename to `rootfs-tools-test.ext2`.